### PR TITLE
Add script to setup nightly builds

### PR DESCRIPTION
**Pull request by pschafhalter**
## What do these changes do?

Adds script to set up nightly builds. Note: script is missing functionality to upload to PyPI (cc @richardliaw).

## Related issue number

 #3798

### CODE REVIEW

File: .travis/upload-nightly-wheels.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+WHEEL_DIR="./.whl"
+mkdir -p $WHEEL_DIR
+
+# Clone Ray to get commit hash
+git clone https://github.com/ray-project/ray.git
+
+# Sync newest wheels from AWS
+COUNT=0
+while ! [[ $WHEELS ]]; do
```
Comment (robertnishihara) [inline .travis/upload-nightly-wheels.sh]: I think we want to do this until all **10** wheels are present (e.g., what if the Linux wheels failed to build).

File: .travis/upload-nightly-wheels.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
```
Comment (robertnishihara) [inline .travis/upload-nightly-wheels.sh]: What will happen in the case where we don't merge any commits for 24 hours? I guess we'll upload a new nightly wheel from the same commit as before?

File: .travis/upload-nightly-wheels.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    popd
+
+    # Attempt to sync wheels
+    aws s3 sync --no-sign-request "s3://ray-wheels/$COMMIT_HASH/" $WHEEL_DIR
+    WHEELS=$(ls $WHEEL_DIR)
+
+    ((COUNT++))
+done
+
+# Rename wheels to ray_nightly-{version}.dev{date}-{info}.whl
```
Comment (robertnishihara) [inline .travis/upload-nightly-wheels.sh]: Will these be ordered properly based on upload date? That is, will `pip install ray-nightly` definitely get the most recent one?

File: .travis/upload-nightly-wheels.sh (hunk 1 above)
Comment (richardliaw) [reply .travis/upload-nightly-wheels.sh]: +1

File: .travis/upload-nightly-wheels.sh (hunk 2 above)
Comment (richardliaw) [reply .travis/upload-nightly-wheels.sh]: Yeah, and that shouldn't be a problem. I guess you probably want a failsafe that prevents you from uploading a commit from before the current live wheels.

File: .travis/upload-nightly-wheels.sh (hunk 3 above)
Comment (richardliaw) [reply .travis/upload-nightly-wheels.sh]: Not sure; we have to test this out (but seems like that's the case with tf_nightly).

**Comment by richardliaw** [comment]
I think we don't need this anymore due to `dev` naming

**Pull request closed without merging**